File: src/oldmaid/OldMaid.java
package oldmaid;

public class OldMaid {
    public static void main(String args[]) {
        Master master = new Master();
        Table field = new Table();

        Player yamamoto = new Player("山本", master, field);
        Player kamei = new Player("亀井", master, field);
        Player saito = new Player("齊藤", master, field);

        master.registerPlayer(yamamoto);
        master.registerPlayer(kamei);
        master.registerPlayer(saito);

        Hand trump = createTrump();

        master.prepareGame(trump);

        master.startGame();
    }

    private static Hand createTrump() {
        Hand trump = new Hand();

        for (int number = 1; number <= 13; number++) {
            trump.addCard(new Card(Card.SUIT_CLUB, number));
            trump.addCard(new Card(Card.SUIT_DIAMOND, number));
            trump.addCard(new Card(Card.SUIT_HEART, number));
            trump.addCard(new Card(Card.SUIT_SPADE, number));
        }

        trump.addCard(new Card(0, Card.JOKER));

        return trump;
    }
}

File: python-oop/src/oldmaid/oldmaid.py
from card import *
from hand import *
from master import *
from player import *
from table import *

# ばば抜きプログラム。


# 53枚のトランプを生成する。
def create_trump():
    trump = Hand([])

    # 各スート13枚のカードを生成する
    for i in range(13):
        number = i + 1
        trump.add_card(Card("SUIT_CLUB", number))
        trump.add_card(Card("SUIT_DIAMOND", number))
        trump.add_card(Card("SUIT_HEART", number))
        trump.add_card(Card("SUIT_SPADE", number))

    # ジョーカーの作成
    trump.add_card(Card(0, 0))

    return trump

# 進行役の生成
master = Master([])

# 場の生成
field = Table()

# プレイヤーの生成
murata = Player("村田", Hand([]), master, field)
yamada = Player("山田", Hand([]), master, field)
saito = Player("斎藤", Hand([]),master, field)

# 進行役へプレイヤーを登録
master.register_player(murata)
master.register_player(yamada)
master.register_player(saito)

# トランプを生成する
trump = create_trump()

# ゲームの準備をする
master.prepare_game(trump)

# ゲームを開始する
master.start_game()
